waypoints_x: [  ]
waypoints_y: [  ]